lib/publish.js

```diff
@@ -22,8 +22,8 @@ module.exports = async (npmrc, {npmPublish, pkgRoot}, pkg, context) => {
     logger.log(`Publishing version ${version} to npm registry on dist-tag ${distTag}`);
     const result = execa(
       'npm',
-      ['publish', basePath, '--userconfig', npmrc, '--tag', distTag, '--registry', registry],
-      {cwd, env, preferLocal: true}
+      ['publish', '--userconfig', npmrc, '--tag', distTag, '--registry', registry],
+      {cwd: basePath, env, preferLocal: true}
```

Comment:
i'd like to point out that when we specify tarballDir it doesn't suffer the same problem and it's because it uses exactly this solution in the prepare.js file see: https://github.com/semantic-release/npm/blob/master/lib/prepare.js#L14-L22